mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.removeOrder = async (req, res) => {
+  try {
+    const deletionResult = await Order.findOneAndDelete({ _id: req.params.orderId });
+    if (!deletionResult) {
+      return res.status(404).json({ error: "Order not found" });
+    }
+    res.json({ message: "Order removed successfully" });
```

Comment:
The response object `res` is not checked for null or undefined before calling the `json` method. This can lead to a runtime error if `res` is null or undefined. It's better to add a null check before calling the `json` method.